### PR TITLE
[OpenMP][DeviceRTL] Added omp_get_max_teams() on device

### PR DESCRIPTION
**Pull request by khoing0810**
Added omp_get_max_teams() on device side and wrote a unit test to ensure functionality.

### CODE REVIEW

File: openmp/libomptarget/DeviceRTL/src/State.cpp [hunk 1]
```diff
@@ -202,6 +204,7 @@ void state::ICVStateTy::assertEqual(const ICVStateTy &Other) const {
 
 void state::TeamStateTy::init(bool IsSPMD) {
   ICVState.NThreadsVar = 0;
+  ICVState.NTeamsVar = 0;
```
Comment (shiltian) [inline openmp/libomptarget/DeviceRTL/src/State.cpp:207]: I don't see how this var is updated among different target regions, so is it necessary to make it part of the team state? Also, I thought this value should be related to the nested level.

Comment (jdoerfert) [reply openmp/libomptarget/DeviceRTL/src/State.cpp:207]: I don't think right now we can update it on the device. That said, once teams doesn't have to be nested strictly, one might be able to set it. I'm fine with an ICV for now. The initialization, via OMP_NUM_TEAMS_DEV_<N>, is going to be a follow up.

Comment (shiltian) [reply openmp/libomptarget/DeviceRTL/src/State.cpp:207]: If that is the case, do we need to put it into team state?

**Comment by shiltian** [comment]
What's going on here? Why was the patch merged w/o an approval?

**Comment by khoing0810** [comment]
i think this is due to me syncing my fork and deleted the previous commit?

**Comment by shiltian** [comment]
Not sure, but you indeed didn't push anything to LLVM's main branch, which is a good thing, but your changes are completely gone now.

**Comment by khoing0810** [comment]
i see. i still have the commit locally and plan to push it again once i update with the  OMP_NUM_TEAMS_DEV_<N> follow-up implementation. if that's not recommended/appropriate for the git/general workflow, please let me know!

**Pull request closed without merging**